Achievement

JesusRivera00 opened the pull request:
This pull request introduces a new feature for tracking user activities and achievements related to recipes. The main changes include the addition of new routes, controllers, and model functions to handle these activities and achievements.

### New Feature: User Activities and Achievements

* **Controller Changes:**
  * [`src/controllers/activityController.ts`](diffhunk://#diff-7de4849ad82adc09d4ed63ebe23ae38e022dc425bf1a649713755632fa8ae1aaR1-R31): Added a new controller function `addActivity` to handle adding user activities and unlocking achievements.

* **Model Changes:**
  * [`src/models/activityModel.ts`](diffhunk://#diff-20d04ca4a7c6d6347599798f6bce248cbc183c009d1632f3f482c1e3ef99d102R1-R43): Introduced new functions `addActivityToDb`, `getActivityCount`, and `addAchievementToDb` to interact with the database for user activities and achievements.

* **Route Changes:**
  * [`src/routes/activityRoutes.ts`](diffhunk://#diff-c52315bf1baff33d9f69763d4a3a4b5d3bd9442daba42803fce426dec4920753R1-R8): Created a new route to handle activity-related requests, specifically to add activities.
  * [`src/server.ts`](diffhunk://#diff-8a8ae07582c9d433ec8c2e5c4310ff8901e604f4965c5b90a49117ad46c47595R5): Integrated the new `activityRoutes` into the main server file to ensure the new routes are active. [[1]](diffhunk://#diff-8a8ae07582c9d433ec8c2e5c4310ff8901e604f4965c5b90a49117ad46c47595R5) [[2]](diffhunk://#diff-8a8ae07582c9d433ec8c2e5c4310ff8901e604f4965c5b90a49117ad46c47595R39)

* **Utility Changes:**
  * [`src/utils/achievement.ts`](diffhunk://#diff-7b716a1942870e799f22d5dbab2822c8d0eaba1d3569fc1522dd9f7e4191e4d6R1-R63): Added a utility function `addActivity` that was refactored to handle adding activities and achievements.

JesusRivera00 committed:
feat: add activity tracking and achievement system

JesusRivera00 committed:
Merge branch 'main' of https://github.com/Arcane-Kitchen/server into achievement

KSmiyazawa commented on src/utils/achievement.ts:
Is this helper function still necessary?

KSmiyazawa commented on src/routes/userRoutes.ts:
```diff
@@ -10,6 +10,7 @@ import {
   updateRecipeInMealPlanByDateAndMealType,
   getUserMealPlan
 } from "../controllers/mealPlanController.js";
+import { addActivity } from "../utils/achievement.js";
```
Are you still going to use `addActivity` in the user routes? If not, please delete. If so, please change it to the activity controller function

JesusRivera00 committed:
Merge branch 'main' of https://github.com/Arcane-Kitchen/server into achievement

JesusRivera00 committed:
feat: add fetchUserAchievements endpoint and remove unused achievement utility

JesusRivera00 committed:
fix: enforce number type for userId and recipeId in activity functions

JesusRivera00 committed:
refactor: remove achievement handling from addActivity and clean up activity model

JesusRivera00 committed:
Merge branch 'main' of https://github.com/Arcane-Kitchen/server into achievement

JesusRivera00 commented:
This pull request introduces new functionality to handle user activities and achievements in the application. The changes include adding new controllers, models, and routes to support these features.

New functionality for handling activities and achievements:

* [`src/controllers/activityController.ts`](diffhunk://#diff-7de4849ad82adc09d4ed63ebe23ae38e022dc425bf1a649713755632fa8ae1aaR1-R54): Added `addActivity` and `fetchUserAchievements` functions to handle adding user activities and fetching user achievements.
* [`src/models/activityModel.ts`](diffhunk://#diff-20d04ca4a7c6d6347599798f6bce248cbc183c009d1632f3f482c1e3ef99d102R1-R44): Implemented `addActivityToDb`, `getActivityCount`, and `getUserAchievements` functions to interact with the database for user activities and achievements.

Routing updates:

* [`src/routes/activityRoutes.ts`](diffhunk://#diff-c52315bf1baff33d9f69763d4a3a4b5d3bd9442daba42803fce426dec4920753R1-R8): Created new routes for adding activities and fetching user achievements.
* [`src/server.ts`](diffhunk://#diff-8a8ae07582c9d433ec8c2e5c4310ff8901e604f4965c5b90a49117ad46c47595R5): Imported `activityRoutes` and added the `/activity` route to the server configuration. [[1]](diffhunk://#diff-8a8ae07582c9d433ec8c2e5c4310ff8901e604f4965c5b90a49117ad46c47595R5) [[2]](diffhunk://#diff-8a8ae07582c9d433ec8c2e5c4310ff8901e604f4965c5b90a49117ad46c47595R38)
